python/pyspark/sql/session.py

```diff
@@ -1024,7 +1024,18 @@ def prepare(obj: Any) -> Any:
         if isinstance(data, RDD):
             rdd, struct = self._createFromRDD(data.map(prepare), schema, samplingRatio)
         else:
-            rdd, struct = self._createFromLocal(map(prepare, data), schema)
+            if isinstance(data, list):
+                # Wrap each element with a tuple if there is any scalar in the list
+                has_scalar = False
+                for x in data:
+                    if isinstance(x, str) or not hasattr(x, "__len__"):  # Scalar
+                        has_scalar = True
+                        break
```

Comment:
How about:
```py
has_scalar = any(isinstance(x, str) or not hasattr(x, "__len__") for x in data)
```
?

Reply:
Looks much cleaner! I guess Python `any` should return immediately when it detects the **first** `True`.